src/community/generated-geometries/src/main/java/org/geoserver/generatedgeometries/core/longitudelatitude/LongLatGeometryGenerationStrategy.java

```diff
@@ -230,23 +232,42 @@ public Filter convertFilter(FeatureTypeInfo info, Filter filter) {
     @Override
     public Query convertQuery(FeatureTypeInfo info, Query query) {
         Query q = new Query();
-        q.setHints(query.getHints());        
+        q.setHints(query.getHints());
         q.setFilter(convertFilter(info, query.getFilter()));
         LongLatConfiguration configuration = getLongLatConfiguration(info);
         List<String> properties = new ArrayList<>();
         try {
-            properties =
-                    info.getFeatureType()
-                            .getDescriptors()
-                            .stream()
-                            .filter(
-                                    propertyDescriptor ->
-                                            !propertyDescriptor
-                                                    .getName()
-                                                    .toString()
-                                                    .equals(configuration.geomAttributeName))
-                            .map(propertyDescriptor -> propertyDescriptor.getName().toString())
-                            .collect(Collectors.toList());
+            // no fields were sent, use all fields excluding geom field
+            if (query.getPropertyNames() == null) {
```

Comment:
Well done, later improve this by also checking is empty.

Reply:
Actually here if no field is request nothing shoudl probably be done., this shoudl be manually tested.

Reply:
No field scenario appears from Layer Config page when bounds are calculated using data and not SRS, in that case the property names come empty